### ISSUE TITLE
Image artifacts with diffuse luminance masking

### ISSUE THREAD

**Issue by groutr** [darktable-org/darktable#10116]
**Describe the bug/issue**
Setting the luminance masking threshold in diffuse module produces a very high intensity noise artifact on the image.

**To Reproduce**
_Please provide detailed steps to reproduce the behaviour, for example:_
1. Pick any image (preferably without any edit history)
2. Activate diffuse or sharpen
3. Set the "luminance masking threshold" to 1.0

**Expected behavior**
I expect to see no artifacts. There is something special about the value 1.0. No other value on the slider seems to produce any artifacts. This happens with/without OpenCL.

**Screenshots**
![diffuselumthresh](https://user-images.githubusercontent.com/10678620/135204332-278e4e35-4701-4571-b0a7-d5729af11cd2.png)

**Platform**
this is darktable 3.7.0+1154~gd3d26d2e3a
copyright (c) 2009-2021 johannes hanika
darktable-dev@lists.darktable.org

compile options:
  bit depth is 64 bit
  normal build
  SSE2 optimized codepath enabled
  OpenMP support enabled
  OpenCL support enabled
  Lua support enabled, API version 8.0.0-dev
  Colord support enabled
  gPhoto2 support enabled
  GraphicsMagick support enabled
  ImageMagick support disabled
  OpenEXR support enabled

**Comment by wpferguson** [comment]
I can reproduce.  At first I thought it was just chroma noise, but lower ISO images are worse than high ISO.  It's the worst at 1% and gradually decreases until it disappears.

**Comment by flannelhead** [comment]
I wonder if this is a bug at all. The module is intended to fill the parts above the luminance threshold with noise (to have initial data for inpainting the highlights). Setting the threshold to 1 % essentially tells the module fill almost all pixels with noise. Can @aurelienpierre chime in here?

One thing I wonder though: the intensity of the noise might be a bit harsh. At least I have never been able to dial in such settings that any number of iterations would "eliminate" the noise completely, filling the selected parts with inpainted colors. But this may be a topic for another issue.

**Comment by aurelienpierre** [comment]
> I wonder if this is a bug at all. The module is intended to fill the parts above the luminance threshold with noise (to have initial data for inpainting the highlights). Setting the threshold to 1 % essentially tells the module fill almost all pixels with noise. Can @aurelienpierre chime in here?

Exactly. We need to create "particles" before starting the diffusion process when we want to perform inpainting. That's achieved with noise.

> One thing I wonder though: the intensity of the noise might be a bit harsh. 

I have tested different noise levels, and it's hard to hit a good generic setting. Perhaps this means we need to expose the noise sigma to the user (but one more slider in this already crowded module…).
